lib/SIL/SILConstants.cpp

```diff
@@ -0,0 +1,395 @@
+SymbolicValue
+SymbolicValue::cloneInto(llvm::BumpPtrAllocator &allocator) const {
+  auto thisRK = representationKind;
+  switch (thisRK) {
+  case RK_Unknown:
+  case RK_Metatype:
+  case RK_Function:
+  case RK_IntegerInline:
+  case RK_Integer:
+    return SymbolicValue::getInteger(getIntegerValue(), allocator);
+  case RK_Aggregate: {
+    auto elts = getAggregateValue();
+    SmallVector<SymbolicValue, 4> results;
+    results.reserve(elts.size());
+    for (auto elt : elts)
+      results.push_back(elt.cloneInto(allocator));
+    return getAggregate(results, allocator);
+  }
+  }
+}
+
+//===----------------------------------------------------------------------===//
+// Integers
+//===----------------------------------------------------------------------===//
+
+SymbolicValue SymbolicValue::getInteger(int64_t value, unsigned bitWidth) {
+  SymbolicValue result;
+  result.representationKind = RK_IntegerInline;
+  result.value.integerInline = value;
+  result.aux.integer_bitwidth = bitWidth;
+  return result;
+}
+
+SymbolicValue SymbolicValue::getInteger(const APInt &value,
+                                        llvm::BumpPtrAllocator &allocator) {
+  // In the common case, we can form an inline representation.
+  unsigned numWords = value.getNumWords();
+  if (numWords == 1)
+    return getInteger(value.getRawData()[0], value.getBitWidth());
+
+  // Copy the integers from the APInt into the bump pointer.
+  auto *words = allocator.Allocate<uint64_t>(numWords);
+  std::uninitialized_copy(value.getRawData(), value.getRawData() + numWords,
+                          words);
+
+  SymbolicValue result;
+  result.representationKind = RK_Integer;
+  result.value.integer = words;
+  result.aux.integer_bitwidth = value.getBitWidth();
+  return result;
+}
+
+APInt SymbolicValue::getIntegerValue() const {
+  assert(getKind() == Integer);
+  if (representationKind == RK_IntegerInline) {
+    auto numBits = aux.integer_bitwidth;
+    return APInt(numBits, value.integerInline);
+  }
+
+  assert(representationKind == RK_Integer);
+  auto numBits = aux.integer_bitwidth;
+  auto numWords = (numBits + 63) / 64;
+  return APInt(numBits, {value.integer, numWords});
+}
+
+unsigned SymbolicValue::getIntegerValueBitWidth() const {
+  assert(getKind() == Integer);
+  assert (representationKind == RK_IntegerInline ||
+          representationKind == RK_Integer);
+  return aux.integer_bitwidth;
+}
+
+//===----------------------------------------------------------------------===//
+// Aggregates
+//===----------------------------------------------------------------------===//
+
+/// This returns a constant Symbolic value with the specified elements in it.
+/// This assumes that the elements lifetime has been managed for this.
+SymbolicValue SymbolicValue::getAggregate(ArrayRef<SymbolicValue> elements,
+                                          llvm::BumpPtrAllocator &allocator) {
+  // Copy the integers from the APInt into the bump pointer.
```

Comment:
@marcrasi The comments are referring to ints. Probably this should be updated for arrays.